src/redux/create-module.js

```diff
@@ -143,12 +150,31 @@ function createModule(namespace, createDefinition) {
   }, {})
 
   // ----------------------------------
+  // Selectors
+  // ----------------------------------
+  // Generate selectors that provide minimal validation help.
+  generatedModule.selectors = Object.keys(definition.selectors).reduce((acc, selectorName) => {
```

Comment:
Like the idea, but personally not a _huge_ fan of runtime type checking. Our existing invariants are there solely at module instantiation time, and after that point the module functions normally. I would much prefer to see us pursue a legitimate type checking solution for things like this than to wrap functions with assertions that don't cover all invariants (what if state is undefined? does it return what it's expected to?).

Not a blocker, but hopefully the distinction between this and interface conformance checks is clear and that it does not come across as a nitpick.